core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -570,6 +572,46 @@ public ServerBuilder maxConnectionAge(Duration maxConnectionAge) {
         return maxConnectionAgeMillis(requireNonNull(maxConnectionAge, "maxConnectionAge").toMillis());
     }
 
+    /**
+     * Sets the connection drain duration in micros for the connection shutdown.
+     * At the beginning of the connection drain server signals the clients that the connection shutdown is
+     * imminent but still accepts in flight requests.
+     * After the connection drain end server stops accepting new requests.
+     * Also, see {@link ServerBuilder#connectionDrainDuration(Duration)}.
+     *
+     * <p>
+     * Note that HTTP/1 doesn't support draining as described here, so for HTTP/1 drain duration
+     * is always {@code 0}.
+     * </p>
+     *
+     * @param durationMicros the drain duration. {@code 0} or negative value disables the drain.
```

Comment:
nit:
```suggestion
     * @param durationMicros the drain duration. {@code 0} disables the drain.
```
.. because we raise an exception if it's a negative value. 😉

Reply:
see https://github.com/line/armeria/pull/3715/files#r677173583

Reply:
Hmm, in this method, we raise an `IllegalArgumentException` if a negative value is specified https://github.com/line/armeria/pull/3715/files#diff-7a02734685159ba698f82fedf9ceaff74bf69725e2836e98f763b51061040d4aR590 which means a negative value does not disable the drain.
So I thought we need to rephrase the sentence a bit. 😉

Reply:
🤦  my bad :)